sources/model/Cloud.cpp

```diff
@@ -173,6 +210,118 @@ void Cloud::setActiveState(bool activateState)
     isActive = activateState;
 }
 
+void Cloud::setActiveMidiState(bool activateMidiState, int l_midiNote, int l_midiVelo)
+{
+    //debug: std::cout << "entree setactivatemidistate, activateMidiState =" << activateMidiState<< ",l_midiNote="<<l_midiNote<<",l_midiVelo="<<l_midiVelo <<std::endl;
+
+    cleanMidiClouds();
+    if (activateMidiState){
+        // midi note on
+        /*if (playedCloudMidi[l_midiNote]->isActive){
+            std::cout << "isActive"<<std::endl;
+            for (int j = 0; j < playedCloudMidi[l_midiNote]->myGrains.size(); j++)
+                std::cout << "boucle suppression, j="<<j<<"/"<<playedCloudMidi[l_midiNote]->myGrains.size()<<std::endl;
+                playedCloudMidi[l_midiNote]->myGrains.pop_back();
+        }
+        for (int j = 0; j < myGrains.size(); j++)
+            std::cout << "boucle creation, j="<<j<<"/"<<playedCloudMidi[l_midiNote]->myGrains.size()<<std::endl;
+            playedCloudMidi[l_midiNote]->myGrains.push_back(new Grain(theSamples, duration, pitch));*/
+        // calculate pitch
+        int ecartNotes = l_midiNote - midiNote;
+        float musicalPitch = (12*log2(pitch));
+        float newMusicalPitch = musicalPitch + ecartNotes;
+        float newPitch = (pow(2, (float) (newMusicalPitch / 12)));
```

Comment:
Fonction math : même remarque.
```suggestion
        float newPitch = (powf(2, (float) (newMusicalPitch / 12)));
```

Reply:
vu